syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.iotservice";
option java_outer_classname = "IoTServiceProto";
option objc_class_prefix = "TEMP";

package iot_service;

// The temperature service definition.
service IoTService {
  // Responds with a temperature measurement
  rpc SayTemperature (TemperatureRequest) returns (TemperatureReply) {}
  
  // Send a command to the led
  rpc BlinkLed (LedRequest) returns (LedReply) {}

  // List all devices
  rpc GetDevices (DeviceRequest) returns (DeviceReply) {}
}

message DeviceRequest {
}

message DeviceReply {
  repeated DeviceItem devices = 1;
}

message DeviceItem {
  string deviceId = 1;
  string state = 2;
  string name = 3;
  string type = 4;
}

// The request message containing the user's name.
message TemperatureRequest {
  string sensorName = 1;
}

// The response message containing the greetings
message TemperatureReply {
  string temperature = 1;
}

message LedRequest {
  string deviceId = 1;
  string state = 2;
}

message LedReply {
  map<string, string> device = 1;
}
